typing/typedtree.mli

```diff
@@ -49,6 +49,7 @@ and 'a pattern_data =
     pat_loc: Location.t;
     pat_extra : (pat_extra * Location.t * attributes) list;
     pat_type: Types.type_expr;
+    pat_mode: Types.alloc_mode;
```

Comment:
What's this for?

Reply:
This mean you can tell what mode a variable is bound at by looking at the typedtree. We use that for checking what escapes in top-level bindings, but seems a useful property in general for the typedtree.